server/study_server.js

```diff
@@ -132,20 +133,43 @@ module.exports = {
                 Response.find({ studyID: req.params.id}, function(err,responses) {
                     if (err) {
                         req.status(504);
-                        logger.error("response_server.js: Cannot find study responses to delete:", error);
+                        logger.error("response_server.js: Cannot find study responses to delete:", err);
                         req.end();
                     } else {
                         for (var i = 0; i < responses.length; i++) {
-                            responses[i].remove();
+                             Event.find({ responseId: responses[i]._id}, function(err,events)
+                             {
+                                if (err) {
+                                    req.status(504);
+                                    logger.error("response_server.js: Cannot find study events to delete:", err);
+                                    req.end();
+                                }
+                                else {
+                                    events[i].deleteOne(function(err){
+                                    if(err) {
+                                        logger.error("response_server.js: Error trying to delete event:", err);
+                                        res.status(500).send(err);
+                                        }
+                                    })
+                                } 
+                            });                       
+                            
+                            responses[i].deleteOne(function(err){
+                                if(err) {
+                                    logger.error("response_server.js: Error trying to delete response:", err);
+                                    res.status(500).send(err);
+                                }
+                            });
                         }
                     }
+                });         
+
+                study.deleteOne(function(err){
+                    if(err) {
+                        logger.error("study_server.js: Error trying to delete study:", err);
+                        res.status(500).send(err);
+                    }
                 });
-            }
-        }).remove(function (err) {
-            if (err) {
-                logger.error("study_server.js: Error, cannot remove study:", err);
-                res.end(err);
-            } else {
                 res.send(true);
```

Comment:
Is it possible to log deletion was a success as well.

Reply:
I can try add a logger at the end to say that it was successful. Just as we have multiple events and responses so it might clog up with logs if we did it for every one of those

Reply:
Good point.

Reply:
Couldn't find a neat way to do a console log in the server without doing a logger.error which i don't think is right for a confirmation log

Reply:
I think you should be able to do `logger.info("...")`

Reply:
I'll give that a try

Reply:
So cannot find where logger.info outputs to. Tried console in vs code, Tried network in dev tools, Tried console in dev tools. Info is not being logged. I mean the study does immediately disappear from UI so delete is visibly successful

Reply:
Should be the same place as logger.error. Unless the logger created in logger.js is not logging info events. The docs say the default level is "info", so "info" and above should be logged.